### PR TITLE
FDA comment Table Primary

### PR DESCRIPTION
**Pull request by SHAESEN2**
1. Updated headers with defensive coding

2. No rounding issue found. CI matched with FDA when I run the code. Perhaps depends on the version of the packages? I was not able to run the renv setup on our company server.

### CODE REVIEW

**Comment by elong0527** [comment]
Please also update pdf file in https://github.com/RConsortium/submissions-pilot1/tree/main/vignettes/pdf

Please share your session information. I still see the difference in PDF version 

https://github.com/RConsortium/submissions-pilot1/blob/main/vignettes/pdf/tlf-primary.pdf

https://github.com/RConsortium/submissions-pilot1/blob/main/vignettes/pdf/tlf-primary.pdf

**Comment by mstackhouse** [comment]
@SHAESEN2 can you give your `SessionInfo()` output? I'm curious about the versions of everything used and I'd like to see if I can replicate myself.

File: vignettes/tlf-primary.Rmd
Comment (mstackhouse) [inline vignettes/tlf-primary.Rmd:82]: @SHAESEN2 this is already defensively coded in Tplyr:

```r
 paste0("|Placebo\\line(N=**Placebo**)| Xanomeline High Dose\\line(N=**Xanomeline High Dose**) ", 
           "| Xanomeline Low\\line Dose\\line(N=**Xanomeline Low Dose**)")
```

The `**Xanomeline High Dose**, **Placebo**` etc. text automatically captures the header N's as calculated when the table is built. 

The output in this case was accurate, unless I'm missing something. Was there another specific comment you're trying to work around, besides the fact that I just had the order of the headers mixed up in my original output?

**Review by mstackhouse — CHANGES_REQUESTED** [review]

File: vignettes/tlf-primary.Rmd
Comment (SHAESEN2) [reply vignettes/tlf-primary.Rmd:82]: Hey @mstackhouse I think my fork was behind this repo, because I did not see this update. The defensive part is that we should not hard code headers, but extract them from the data so that these mistakes don't occur.

**Comment by SHAESEN2** [comment]
> @SHAESEN2 can you give your `SessionInfo()` output? I'm curious about the versions of everything used and I'd like to see if I can replicate myself.

R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server 7.9 (Maipo)

Matrix products: default
BLAS:   /usr/lib64/libblas.so.3.4.2
LAPACK: /usr/lib64/liblapack.so.3.4.2

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.6.0 tools_3.6.0    knitr_1.33     xfun_0.25      renv_0.14.0

File: vignettes/tlf-primary.Rmd
Comment (mstackhouse) [reply vignettes/tlf-primary.Rmd:82]: Ok I see what you're doing now. Sorry - at first I interpreted as you calculating the header N's yourself because I didn't look at the code closely enough.

Thanks - this would be a good update to provide tooling in Tplyr as well.

**Comment by mstackhouse** [comment]
@elong0527 @SHAESEN2 - the rounding difference is a change between R 3.6  and R 4.0:

# R 3.6
```r
> round(.65, digits=1)
[1] 0.6
```

# R 4.1
```r
> round(.65, digits=1)
[1] 0.7
```

**Comment by mstackhouse** [comment]
@SHAESEN2 I'll flip this to approve and merge once Yilong's comments [here](https://github.com/RConsortium/submissions-pilot1/pull/19#issuecomment-1016579364) are updated.

**Comment by SHAESEN2** [comment]
Thanks!!!

**Comment by elong0527** [comment]
There are two more comments from reviewer in 

https://github.com/RConsortium/submissions-wg/issues/61#issuecomment-1018047645

Shall we align if we need to address any of them? 

1. In tlf-primary.r the last hf_line for the technical footnote yielded date/time in local language and not english. Again maybe just my machine (windows with local R 4.1.2).
2. treatment/column ordering is not consistent (We might consider Placebo, low does, high does, then both `tlf-primary` and `kmplot` need to be updated.)

**Comment by elong0527** [comment]
I could handle those PDF files after this PR merged. It was used to preview results in the pkgdown website.

**Comment by mstackhouse** [comment]
@elong0527 for point number 1, can you show me what you're getting? I have to assume it's a local language thing because I've never tested for other langauge configurations.

For 2, somehow I messed up the order of this, and I'll fix it. I _thought_ I was matching the original output but apparently I mixed it up at some point. The original pilot goes placebo, low, high. So I think we should do that. 

Great example of why we typically double program in industry :)

**Comment by mstackhouse** [comment]
@SHAESEN2 I don't know if I'm doing something wrong here, but I can't see this branch on your fork. Looking at your fork, I only see the same branches as this repo. 

I'm happy to do the updates to fix the column ordering. If I can grab your branch directly,  I can do it here. Otherwise, we can merge this branch and I can do a separate PR.

**Comment by SHAESEN2** [comment]
> @SHAESEN2 I don't know if I'm doing something wrong here, but I can't see this branch on your fork. Looking at your fork, I only see the same branches as this repo.
> 
> I'm happy to do the updates to fix the column ordering. If I can grab your branch directly, I can do it here. Otherwise, we can merge this branch and I can do a separate PR.

I had to delete my original fork because I messed it up :( Could you merge this branch and create a separate PR? Let me know if you need help.

**Comment by elong0527** [comment]
I merged it to `tlf-primary` branch. So we can work from there.

**Comment by NNaikp** [comment]
> @NNaikp could you share your output on timestamp and R language setting? Thanks!

Github doesn't support .rtf so I have attached the file here ([tlf-primary.docx](https://github.com/RConsortium/submissions-pilot1/files/7937055/tlf-primary.docx)) as .docx. Here is also a screenshot from the .rtf file:

image

My default language in Word is UK english.

When I did the review I set `Sys.setenv(LANG = "en")` as it was showing me error messages in danish. But after some googling I can see that the locale is still Danish, also if I set the language to English:

```{R}
> Sys.setenv(LANG = "en")
> Sys.getlocale()
[1] "LC_COLLATE=Danish_Denmark.1252;LC_CTYPE=Danish_Denmark.1252;LC_MONETARY=Danish_Denmark.1252;LC_NUMERIC=C;LC_TIME=Danish_Denmark.1252"
```

After running `Sys.setlocale(locale = "English")` I see
```{R}
> Sys.setlocale(locale = "English")
[1] "LC_COLLATE=English_United States.1252;LC_CTYPE=English_United States.1252;LC_MONETARY=English_United States.1252;LC_NUMERIC=C;LC_TIME=English_United States.1252"
```

Which yields:

image

Maybe `pharmaRTF::write_rtf` needs to check the locale before sinking the file? I can see from the documentation of `Sys.setlocale()` that it is OS dependent. The one above is Windows (my local). This one is from a linux server:

```{R}
> Sys.getlocale()
[1] "LC_CTYPE=en_US.UTF-8;LC_NUMERIC=C;LC_TIME=en_US.UTF-8;LC_COLLATE=en_US.UTF-8;LC_MONETARY=en_US.UTF-8;LC_MESSAGES=en_US.UTF-8;LC_PAPER=en_US.UTF-8;LC_NAME=C;LC_ADDRESS=C;LC_TELEPHONE=C;LC_MEASUREMENT=en_US.UTF-8;LC_IDENTIFICATION=C"
```

**Comment by mstackhouse** [comment]
@NNaikp thanks and I logged an issue in pharmaRTF right [here](https://github.com/atorus-research/pharmaRTF/issues/8#issue-1114382989)

**Pull request merged**
